### ISSUE TITLE
Unique column in kundera mongodb

### ISSUE THREAD

**Issue by sreekanth0041** [Impetus/kundera#697]
I want restaurant_item_name must be Unique

My class is below

```
@Entity
@Table(name = "restaurant_item", schema = "xxx@xxx")
public class RestaurantItem implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private String restaurant_item_id;

    @Column(name = "restaurant_item_name")
    private String restaurant_item_name;
//setter and getters
}
```

Help me.

**Comment by chhavigangwal** [comment]
@sreekanth0041 

Currently you cannot specify this using using Kundera, however you can ensure unique names in  MongoDB  by creating a unique index over it in the following manner :

```
db.restaurant_item.ensureIndex( { restaurant_item_name: 1 }, { unique: true } )
```

Hope this Helps!
Chhavi

**Comment by sreekanth0041** [comment]
Thank you for giving the replay chhavigangwal,We need to do it manually right?

**Comment by chhavigangwal** [comment]
Yes

Chhavi

**Comment by sreekanth0041** [comment]
Ok thank you

**Comment by mevivs** [comment]
@sreekanth0041

http://docs.mongodb.org/manual/tutorial/create-a-unique-index/

Better to use 

```
db.restaurant_item.ensureIndex( { restaurant_item_name: 1 }, { unique: true, sparse: true } )
```

Also, IMHO it should be invoked during application bootstrap.  With Kundera, i guess you should be able to execute it in native way. So while instantiating EMF, get a handle of Mongo DB client and just invoke it.

Kundera Team, can enable this support by extending @Index and @IndexCollection for index uniqueness.

HTH,
-Vivek

**Comment by chhavigangwal** [comment]
Fixed with Kundera 3.0 release.
